lib/solveEquation/stepThrough.js

```diff
@@ -183,11 +221,18 @@ function addSimplificationSteps(steps, equation, debug=false) {
   let oldEquation = equation.clone();
 
   const leftSteps = simplifyExpressionNode(equation.leftNode, false);
+  const leftFactorSteps = factor(equation.leftNode, false);
   const leftSubSteps = [];
   for (let i = 0; i < leftSteps.length; i++) {
     const step = leftSteps[i];
     leftSubSteps.push(EquationStatus.addLeftStep(equation, step));
   }
+
+  for (let i = 0; i < leftFactorSteps.length; i++) {
```

Comment:
actually I think this works. But you should dig in a bit and understand how it works and add comments explaining it - especially if you keep working in this area of the code you should know what's going on!